### DISCUSSION TITLE
Rust version 1.70 and tokio version compatibility

### DISCUSSION THREAD

**Discussion by deepa2015** [tokio-rs/tokio#6391] [General] [answered]
Hello,
I'm new to RUST and trying to build tokio sample using rust 1.70. However, get build errors.

Source and build error logs are attached.
Please help fix the build error.

Which tokio version works with Rust 1.70 ?

Thanks
[build-error.log](https://github.com/tokio-rs/tokio/files/14562090/build-error.log)
[Cargo.lock.txt](https://github.com/tokio-rs/tokio/files/14562102/Cargo.lock.txt)
[Cargo.toml.txt](https://github.com/tokio-rs/tokio/files/14562103/Cargo.toml.txt)
[main.rs.txt](https://github.com/tokio-rs/tokio/files/14562104/main.rs.txt)

**Comment by Darksonn** [comment, marked as answer]
All versions of Tokio are compatible with Rust 1.70.

Looking at your build error, it looks like there is something wrong with the libc crate. What platform are you building for?

**Comment by deepa2015** [comment]
It is built for x86_64 , proprietary RTOS .

Thanks

**Reply by Darksonn** [reply]
Most likely the issue is that the libc crate doesn't support the target you are compiling for.

**Comment by deepa2015** [comment]
Are there any references or documents to port tokio to new target / platform?
Thanks

**Reply by Darksonn** [reply]
No, we do not have documentation on that.

**Reply by deepa2015** [reply]
The question is naive as I'm new to RUST.
Is there any quick way to identify  if the crate(or dependent crates) need platform specific modifications? Thanks
